flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBStateUploader.java

```diff
@@ -80,9 +82,9 @@ public Map<StateHandleID, StreamStateHandle> uploadFilesToCheckpointFs(
         try {
             FutureUtils.waitForAll(futures.values()).get();
 
-            for (Map.Entry<StateHandleID, CompletableFuture<StreamStateHandle>> entry :
+            for (Map.Entry<String, CompletableFuture<StreamStateHandle>> entry :
                     futures.entrySet()) {
-                handles.put(entry.getKey(), entry.getValue().get());
+                handles.add(HandleAndLocalPath.of(entry.getValue().get(), entry.getKey()));
```

Comment:
The code in `uploadFilesToCheckpointFs` can be simplified if `HandleAndLocalPath` is created inside `createUploadFutures`.

Reply:
Good suggestion, done.